Module4.3_Search_and_Sorting_Algorithms/activities/Act3_BinaryHeaps/2.md

```diff
@@ -14,7 +14,33 @@ The next method we will implement is `insert`. The easiest, and most efficient,
 
 Notice that when we percolate an item up, we are restoring the heap property between the newly added item and the parent. We are also preserving the heap property for any siblings. Of course, if the newly added item is very small, we may still need to swap it up another level. In fact, we may need to keep swapping until we get to the top of the tree. 
 
-Below depicts the `percUp` method, which percolates a new item as far up in the tree as it needs to go to maintain the heap property. Here is where our wasted element in `heapList` is important. Notice that we can compute the parent of any node by using simple integer division. The parent of the current node can be computed by dividing the index of the current node by 2.
+Let's look at the `percUp` method, which percolates a new item as far up in the tree as it needs to go to maintain the heap property. 
+
+We pass one argument, `i`, which is the size of the heap, because that index is where the new value is located. Then, we use a `while` loop, which simply checks that there are enough values in the heap to use this function. 
+
+```python
+def percUp(self,i):
+    while i // 2 > 0:
+```
+
+Next, we enter the while loop. Here is where our wasted element in `heapList` is important. Notice that we can compute the parent of any node by using simple integer division. The parent of the current node can be computed by dividing the index of the current node by 2.
```

Comment:
move this upwards. this can be resolved with the previous comment

Reply:
move the paragraph and the code block up? how far up ?